site/app/controllers/admin/ReportController.php

```diff
@@ -666,62 +704,174 @@ public function generateCustomization() {
         }
     }
 
+
+    #[Route("/courses/{_semester}/{_course}/reports/build_form", methods: ['POST'])]
+    public function executeBuildForm(): JsonResponse {
+        // Configure json to go into jobs queue
+        $job_json = [
+            'job' => 'RunAutoRainbowGrades',
+            'semester' => $this->core->getConfig()->getTerm(),
+            'course' => $this->core->getConfig()->getCourse(),
+        ];
+
+        // Encode
+        $job_json = json_encode($job_json, JSON_PRETTY_PRINT);
+
+        // Create path to new jobs queue json
+        $path = '/var/local/submitty/daemon_job_queue/auto_rainbow_' .
+            $this->core->getConfig()->getTerm() .
+            '_' .
+            $this->core->getConfig()->getCourse() .
+            '.json';
```

Comment:
It would be better to use `FileUtils::joinPaths()` for this.

Reply:
Well, this is more of `.` separator for string concatenation, but sure

Reply:
Oh, my bad on that.  I thought there were slashes between the components.